src/mds/Locker.cc

```diff
@@ -5548,6 +5589,14 @@ void Locker::local_xlock_finish(const MutationImpl::lock_iterator& it, MutationI
   lock->put_xlock();
   mut->locks.erase(it);
 
+  if (lock->get_type() == CEPH_LOCK_IQUIESCE) {
+    auto in = static_cast<CInode*>(lock->get_parent());
+    set<CInode*> issue_set { in };
+    // reevaluate everything related related to caps
+    eval_cap_gather(in, &issue_set);
+    issue_caps_set(issue_set);
```

Comment:
This is one of the key points of this approach. since we alter the allowed caps based on the quiesce lock xlocked state, here we let the system reissue the capabilities that are no longer masked. It happens automatically when the quiesce xlock is dropped along with the outstanding quiesce request.